### PR TITLE
ws-manager: Add the workspace status for PodInitialzing phase

### PR DESCRIPTION
**Pull request by utam0k**
## Description

For some reason, when there are two containers in a pod, there is no ContainerCreating phase, but PodInitializing. So we have addressed this issue.

## Related Issue(s)

Fixes https://github.com/gitpod-io/gitpod/issues/14103

## Release Notes

```release-note
NONE
```

## Documentation

No

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [x] /werft with-large-vm
- [x] /werft with-integration-tests=workspace
      Valid options are `all`, `workspace`, `webapp`, `ide`

### CODE REVIEW

File: .gitleaks.toml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+[allowlist]
```
[the remark this answers is not in the document]
Comment (utam0k) [reply .gitleaks.toml:1]: We cannot pass the precommit hooks without this file because thb golden file has a token and etc.

**Review by jenting — COMMENTED** [review]
> For some reason, when there are two containers in a pod

Isn't it because we want to add initContainer to chown?

**Review by jenting — COMMENTED** [review]
I don't know how to test it because the code change LGTM.

/hold for @sagor999 double check.

**Review by jenting — APPROVED** [review]

**Comment by utam0k** [comment]
> > For some reason, when there are two containers in a pod
> 
> Isn't it because we want to add initContainer to chown?

Yes, it is. Becase a workspace status doesn't hit the Creating phase.

**Comment by sagor999** [comment]
Thank you @utam0k! Can confirm this PR did fix the issue with more then one container in the workspace pod. Good find!

**Pull request merged**